lib/modules.nix

```diff
@@ -563,32 +565,36 @@ rec {
           qux = [ "module.hidden=baz,value=bar" "module.hidden=fli,value=gne" ];
         }
       */
-      byName = attr: f: modules:
+      byName = isConfig: attr: f: modules:
```

Comment:
This makes me doubt that `byName` would be a good abstraction.
Inlining `byName` will likely make the code easier to read and remove a bunch of function calls from the hot path.

Alternatively, you could use `attr == "config"` instead of introducing another lambda. That way at least performance is unaffected by this PR; not introducing a lambda and corresponding calls.